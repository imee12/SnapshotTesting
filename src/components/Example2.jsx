import React from 'react';
import Icon from './Icon'


const Example2 = () => (
  <div>
    <h1>EXAMPLE2</h1>
    <Icon icon="soccer-ball-o" />
  </div>
);

export default Example2;
