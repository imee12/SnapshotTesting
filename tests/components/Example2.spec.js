import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Example2 from '../../src/components/Example2';


describe('Example2', () => {
  test('renders a shallow example', () => {
    const wrapper = shallow(<Example2 />);

    expect(wrapper).toMatchSnapshot();
  });


  test('renders a detailed example with rendered children', () => {
    const component = renderer.create(<Example2 />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
