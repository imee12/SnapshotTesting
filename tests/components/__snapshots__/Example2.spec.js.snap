// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example2 renders a detailed example with rendered children 1`] = `
<div>
  <h1>
    EXAMPLE2
  </h1>
  <span>
    <i
      className="fa fa-soccer-ball-o"
      title=""
    />
    <span
      className="visually-hidden"
    >
      
    </span>
  </span>
</div>
`;

exports[`Example2 renders a shallow example 1`] = `
<div>
  <h1>
    EXAMPLE2
  </h1>
  <Icon
    altText=""
    className=""
    icon="soccer-ball-o"
  />
</div>
`;
